dotCMS/src/integration-test/java/com/dotcms/util/content/json/PopulateContentletAsJSONUtilTest.java

```diff
@@ -0,0 +1,217 @@
+package com.dotcms.util.content.json;
+
+import com.dotcms.IntegrationTestBase;
+import com.dotcms.datagen.ContentletDataGen;
+import com.dotcms.datagen.SiteDataGen;
+import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Logger;
+import org.apache.felix.framework.OSGIUtil;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import static org.junit.Assert.*;
+
+public class PopulateContentletAsJSONUtilTest extends IntegrationTestBase {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+
+        if (!OSGIUtil.getInstance().isInitialized()) {
+            OSGIUtil.getInstance().initializeFramework();
+        }
+    }
+
+    private void removeContentletAsJSONColumn() throws DotDataException {
+
+        try {
+            DbConnectionFactory.getConnection().setAutoCommit(true);
+        } catch (SQLException e) {
+            throw new DotDataException(e.getMessage(), e);
+        }
+
+        final DotConnect dotConnect = new DotConnect();
+        try {
+            dotConnect.setSQL("alter table contentlet drop column contentlet_as_json");
+            dotConnect.loadResult();
+        } catch (DotDataException e) {
+            Logger.error(this, "Error removing contentlet_as_json column", e);
+        }
+    }
+
+    private void createContentletAsJSONColumn() throws DotDataException {
+
+        try {
+            DbConnectionFactory.getConnection().setAutoCommit(true);
+        } catch (SQLException e) {
+            throw new DotDataException(e.getMessage(), e);
+        }
+
+        final String type;
+        if (DbConnectionFactory.isPostgres()) {
+            type = "JSONB";
+        } else {
+            type = "NVARCHAR(MAX)";
+        }
+
+        final DotConnect dotConnect = new DotConnect();
+        try {
+            dotConnect.setSQL(String.format("ALTER TABLE contentlet ADD contentlet_as_json %s", type));
+            dotConnect.loadResult();
+        } catch (DotDataException e) {
+            Logger.error(this, "Error creating contentlet_as_json column", e);
+        }
+    }
+
+    /**
+     * Method to test: {@link PopulateContentletAsJSONUtil#populateForAssetSubType(String)}
```

Comment:
Let's doc these methods based on our new standard:

* Method to test
* Given Scenario
* Expected Result

Reply:
this is the template:
/**
* Method to test: include this in case the method to test does not belong to the class this test-class is testing 
* Given Scenario: Scenario under test
* ExpectedResult: Expected result that the assertion is checking against
*
*/

Reply:
Done